projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -492,27 +493,35 @@ public function render_block( $attributes, $content ) {
 			$wrapper_attributes = \WP_Block_Supports::get_instance()->apply_block_supports();
 		}
 
-		$headline_id     = 'relatedposts-headline-' . uniqid();
 		$headline_markup = '';
 
-		if ( $block_attributes['show_headline'] === true ) {
+		if ( isset( $block ) ) {
+			foreach ( $block->inner_blocks as $inner_block ) {
+				if ( 'core/heading' === $inner_block->name && ! empty( trim( wp_strip_all_tags( $inner_block->inner_html ) ) ) ) {
+					$headline_markup = $inner_block->inner_html;
```

Comment:
Should we add the escaping here instead of in the check above?

Reply:
Makes sense!